megamek/src/megamek/client/bot/princess/WeaponFireInfo.java

```diff
@@ -434,13 +459,36 @@ private WeaponAttackAction buildBombAttackAction(final HashMap<String, int[]> bo
      * @return expected damage of firing a TAG weapon, in light of other options.
      */
     double computeAeroExpectedTAGDamage(){
-        int myWeaponsDamage = Compute.computeTotalDamage(shooter.getTotalWeaponList());
+        // If TAG damage exceeds the attacking unit's own max damage capacity, go for it!
+        return computeExpectedTAGDamage(true);
+    }
+
+    /**
+     * Generalized computation of hitting with TAG given current guidable muniitions in play
+     * @param skipOtherWeapons true if Aero, false otherwise.
+     * @return
+     */
+    double computeExpectedTAGDamage(boolean exclusiveWithOtherWeapons){
+        final StringBuilder msg = new StringBuilder("Assessing the expected max damage from ")
+                .append(shooter.getDisplayName())
+                .append(" using their TAG this turn");
+
+        int myWeaponsDamage = 0;
+        if (exclusiveWithOtherWeapons) {
+            // We need to know what we're giving up if we fire this TAG...
+            myWeaponsDamage = Compute.computeTotalDamage(shooter.getTotalWeaponList());
+            msg.append("\nThe unit will be giving up ")
+                    .append(myWeaponsDamage)
+                    .append(" damage from other weapons");
+        }
 
         // Get a list of incoming or potentially incoming guidable weapons from the relevant Princess and compute damage.
         int incomingAttacksDamage = Compute.computeTotalDamage(owner.computeGuidedWeapons(shooter, target.getPosition()));
+        int utility = incomingAttacksDamage - myWeaponsDamage;
+        msg.append("\n\tUtility: ").append(utility).append(" damage (Max, estimated)");
+        LogManager.getLogger().debug(msg.toString());
```

Comment:
Let's make this conditional or eliminate it, as .debug is a very expensive call relatively speaking